File: benchmark/c/sq.c
#include <stdio.h>
#include "time.h"

int sq(int n) {
  int m = n;
  for(int i = 1; i < n; i++)  {
    m += n;
  }
  return m;
}

int main() {
  int iter = 100;
  int i = 0;
  int s = get_micro_time();
  while (i < iter) {
    int res = sq(500000);
    i++;
  }
  int e = get_micro_time();
  printf("%d\n", (e - s));
  return 0;
}

File: benchmark/mincaml/sq.ml
let rec sq n =
  let rec loop i acc =
    if i = 0 then acc
    else loop (i - 1) (acc + n)
  in loop n 0
in
let s = get_micro_time () in
let rec time i =
  if i = 0 then ()
  else let _ = sq 500000 in time (i - 1)
in
time 100;
let e = get_micro_time () in
print_int (e - s);
print_newline ()
